packages/mui-material/src/Input/inputClasses.ts

```diff
@@ -29,6 +29,10 @@ export interface InputClasses {
   inputSizeSmall: string;
   /** Styles applied to the input element if `multiline={true}`. */
   inputMultiline: string;
+  /** Styles applied to the input element if `startAdornment` is provided. */
+  inputAdornedStart: string;
+  /** Styles applied to the input element if `endAdornment` is provided. */
+  inputAdornedEnd: string;
```

Comment:
If I remove both of these classes, it throws an error in outlined variant textfield, when the `variant="outlined"` is not explicitly provided to the Text Field component.

Also, these classes are for the standard text field. Not sure why it throws type error for an implicit outlined text field if I remove them. If you can find out the reason it would be great.

Reply:
@ZeeshanTamboli when `variant="outlined"` is `not` applied, in component implementaion code, default component is set to `OutlinedInput` [here](https://github.com/mui/material-ui/blob/master/packages/mui-material/src/TextField/TextField.js#L105). but `types` in `d.ts` file doesn't know that if variant is not provided it should apply `OutlinedTextFieldProps` to `TextFieldProps`. 

>but `types` in `d.ts` file doesn't know that if variant is not provided it should apply `OutlinedTextFieldProps` to `TextFieldProps`. 

code reference: https://github.com/mui/material-ui/blob/master/packages/mui-material/src/TextField/TextField.d.ts#L229

 
>  but `types` `d.ts` file doesn't know that if variant is not provided it should apply `OutlinedTextFieldProps` to `TextFieldProps`.

Although this will be fixed in this PR https://github.com/mui/material-ui/pull/36863

Reply:
If the type is going to be fixed in #36737 where TypeScript will be able to pick `OutlinedTextFieldProps` for implicit outlined variant, then I think you can include these classes in that PR. There isn't a need to create a separate PR because it is related to the bug you are trying to solve. Here in this PR, in my opinion, it is confusing for implicit outlined text field (i.e variant is not applied). The classes being added here is only for the standard variant, not for outlined text fields. 

This would help in future if somebody wants to check the related PR. Basically, it should not confuse a future reader. We also don't know when that PR will be merged and released after this.

Reply:
@ZeeshanTamboli okay, i've moved changes to #36737 , can you review that PR